crates/ruff_cli/src/commands/run.rs

```diff
@@ -137,3 +144,85 @@ pub fn run(
 
     Ok(diagnostics)
 }
+
+#[cfg(test)]
+mod test {
+    use std::path::PathBuf;
+    use std::str::FromStr;
+
+    use anyhow::Result;
+    use indoc::formatdoc;
+    use path_absolutize::Absolutize;
+
+    use ruff::fix::FixMode;
+    use ruff::logging::LogLevel;
+    use ruff::resolver::PyprojectDiscovery;
+    use ruff::settings::configuration::{Configuration, RuleSelection};
+    use ruff::settings::flags::{Cache, Noqa};
+    use ruff::settings::types::SerializationFormat;
+    use ruff::settings::AllSettings;
+    use ruff::RuleSelector;
+
+    use crate::args::Overrides;
+    use crate::printer::{Flags, Printer};
+
+    use super::run;
+
+    #[test]
+    fn test_jupyter_notebook_integration() -> Result<()> {
+        let mut overrides: Overrides = Default::default();
+        overrides.select = Some(vec![
+            RuleSelector::from_str("B")?,
+            RuleSelector::from_str("F")?,
+        ]);
+
+        let mut configuration = Configuration::default();
+        configuration.rule_selections.push(RuleSelection {
+            select: Some(vec![
+                RuleSelector::from_str("B")?,
+                RuleSelector::from_str("F")?,
+            ]),
+            ..Default::default()
+        });
+
+        let root_path = PathBuf::from(env!("CARGO_MANIFEST_DIR"))
+            .join("..")
+            .join("ruff")
+            .join("resources")
+            .join("test")
+            .join("fixtures")
+            .join("jupyter");
+
+        let diagnostics = run(
+            &[root_path.join("valid.ipynb")],
+            &PyprojectDiscovery::Fixed(AllSettings::from_configuration(configuration, &root_path)?),
+            &overrides,
+            Cache::Disabled,
+            Noqa::Enabled,
+            FixMode::None,
+        )?;
+
+        let printer = Printer::new(
+            SerializationFormat::Text,
+            LogLevel::Default,
+            FixMode::None,
+            Flags::SHOW_VIOLATIONS,
+        );
+        let mut writer: Vec<u8> = Vec::new();
+        printer.write_once(&diagnostics, &mut writer)?;
+        // TODO: Set jupyter notebooks as none-fixable for now
```

Comment:
Do you mind using the `TODO(konsti)`-style TODOs? I like to have authors indicated in the TODOs, not as a sign that they're responsible for fixing them, but as a guidepost for who wrote it and has context on it.

Reply:
What about git blame? Is it in case of refactoring?